core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -52,7 +52,8 @@ class HistoryServer(
     provider: ApplicationHistoryProvider,
     securityManager: SecurityManager,
     port: Int)
-  extends WebUI(securityManager, securityManager.getSSLOptions("historyServer"), port, conf)
+  extends WebUI(securityManager, securityManager.getSSLOptions("historyServer"),
+    port, conf, poolSize = 1000)
```

Comment:
We can add a bit more comments to explain why history server needs a larger pool size.

Reply:
BTW we can also set the `name` parameter properly as you did before.